aixplain/enums/function.py

```diff
@@ -27,9 +27,53 @@
 from aixplain.utils.request_utils import _request_with_retry
 from enum import Enum
 from urllib.parse import urljoin
+import logging
+import time
+from datetime import datetime, timedelta
+import json
+import os
+
+CACHE_FILE = ".aixplain_cache/functions.json"
+CACHE_DURATION = timedelta(hours=24)
 
+def save_to_cache(functions, functions_input_output):
+    def convert_sets(obj):
+        if isinstance(obj, dict): return {k: convert_sets(v) for k, v in obj.items()}
+        elif isinstance(obj, list): return [convert_sets(i) for i in obj]
+        elif isinstance(obj, set): return list(obj)
+        else: return obj
+    cache_data = {"timestamp": time.time(), "functions": {"enum": {name: value for name, value in functions.__members__.items()}, "input_output": convert_sets(functions_input_output)}}
+    temp_file = CACHE_FILE + ".tmp"
+    try:
+        os.makedirs(os.path.dirname(CACHE_FILE), exist_ok=True); json_str = json.dumps(cache_data); logging.info(f"Serialized JSON size: {len(json_str)} bytes")
```

Comment:
Why everything in the same line?

Reply:
why do we need to serialize the JSON in bytes? Couldn't you save it using:

```
import json
with open(CACHE, "w") as f:
   json.dump(dictionary, f, indent=4)
```

Reply:
When I used dump it cut off abruptly for some reason, I changed to serialize and then right, that worked and the json files looked more readable too.

Reply:
Please investigate this deeply. The outcome is already a JSON as you can see [here](https://github.com/aixplain/aiXplain/blob/main/aixplain/enums/function.py#L44). My guess is that there is not need to change anything.